### DISCUSSION TITLE
Works on Windows?

### DISCUSSION THREAD

**Discussion by KrychuTM** [akinomyoga/ble.sh#194] [Q&A] [answered]
Heya heya, just wanted to know if this works on Windows?

**Comment by akinomyoga** [comment, marked as answer]
It depends on what you mean by *Windows*. `ble.sh` is a configuration for Bash, so it does not work if there are no Bash. `ble.sh` works as far as Bash and other basic POSIX shell utilities are available, i.e., it does not work under the plain Windows without Bash but works in WSL, Cygwin family (including MSYS, MSYS2, and Git for Windows), and other environments that provide Bash and basic POSIX utilities for Windows.

**edit**: To build `ble.sh` and related files from the source, additional utilities `gawk` and `make` (GNU make) are needed, but they can be installed in the above environments using their respective standard package systems. Or you can download the pre-built version of `ble.sh` from [the release page](https://github.com/akinomyoga/ble.sh/releases).
